src/lib.rs

```diff
@@ -30,6 +30,9 @@ pub enum TlaError {
 }
 
 pub fn rewrite(input: &str, mode: &Mode, force: bool) -> Result<String, TlaError> {
+    // if the input ends with '\n', we should put the '\n' back to output
+    let end_of_newline = input.chars().last().map_or(false, |x| x == '\n');
```

Comment:
A better variable name might be:
```suggestion
    let input_ends_in_newline = input.chars().last().map_or(false, |x| x == '\n');
```